server/src/main/java/hu/flowacademy/epsilon/sport_event_organizer/service/MatchService.java

```diff
@@ -0,0 +1,261 @@
+                            notQualificationTeam = team;
+                        } else if (notQualificationTeam.getGoalDifference() > team.getGoalDifference()) {
+                            notQualificationTeam = team;
+                        }
+                    }
+                }
+                for (Team team : group) {
+                    if (team.getWinnerCounter() == 2 && team != notQualificationTeam) {
+                        team.setQualified(true);
+                    }
+                }
+
+            } else {
+                for (Team team : group) {
+                    if (team.getWinnerCounter() == 3 && team.getWinnerCounter() == 2) {
+                        team.setQualified(true);
+                    }
+                    teamService.save(team);
+                }
+            }
+        }
+
+        //meccsek legenerálása
+        cupTeams = new ArrayList<Team>(cupService.getApprovedTeams(cupName));
+        Match cNotQualifiedMatch = new Match();
+        Match dNotQualifiedMatch = new Match();
+        Match eQualifiedMatch = new Match();
+        Match fQualifiedMatch = new Match();
+        List<Match> qualifierMatches = new ArrayList();
+
+        for (Team team : cupTeams) {
+            if (team.isQualified()) {
+                if (team.getGroupName() == "Group1") {
+                    eQualifiedMatch.setTeamA(team);
+                } else {
+                    eQualifiedMatch.setTeamB(team);
+                }
+            } else {
+                if (team.getGroupName() == "Group1") {
+                    cNotQualifiedMatch.setTeamA(team);
+                } else {
+                    cNotQualifiedMatch.setTeamB(team);
+                }
+            }
+//            cupTeams.remove(team);
+        }
+        for (Team team : cupTeams) {
+            if (team.isQualified()) {
+                if (team.getGroupName() == "Group1") {
+                    fQualifiedMatch.setTeamA(team);
+                } else {
+                    fQualifiedMatch.setTeamB(team);
+                }
+            } else {
+                if (team.getGroupName() == "Group1") {
+                    dNotQualifiedMatch.setTeamA(team);
+                } else {
+                    dNotQualifiedMatch.setTeamB(team);
+                }
+            }
+//            cupTeams.remove(team);
+        }
+
+
+        qualifierMatches.add(cNotQualifiedMatch);
+        qualifierMatches.add(dNotQualifiedMatch);
+        qualifierMatches.add(eQualifiedMatch);
+        qualifierMatches.add(fQualifiedMatch);
+        cNotQualifiedMatch.setGroupName("Not Qualified");
+        dNotQualifiedMatch.setGroupName("Not Qualified");
+        eQualifiedMatch.setGroupName("Qualified");
+        fQualifiedMatch.setGroupName("Qualified");
+
+
+        qualifierMatches.forEach(match -> match.setCup(cup));
+
+
+        //pályák szétosztása
+        String courtName = "Court1";
+        for (int i = 0; i < matches.size(); i++) {
```

Comment:
Ezt nem lehetne valamilyen automatikusabb módon generálni. Gondolok itt String konkatenációra.

Reply:
Úgy látom kétszer is szerepel a kódban, ne ismételjétek magatok!